celer/homotopy.py

```diff
@@ -144,16 +144,19 @@ def celer_path(X, y, pb, eps=1e-3, n_alphas=100, alphas=None,
         (Is returned only when ``return_thetas`` is set to True).
     """
 
-    if pb.lower() not in ("lasso", "logreg", "grouplasso"):
+    pb = pb.lower()
+    if pb not in ("lasso", "logreg", "grouplasso"):
         raise ValueError("Unsupported problem %s" % pb)
-    if pb.lower() == "lasso":
+    # prevent referenced before assigned error
+    grp_ptr, grp_indices, n_groups = None, None, None
+    if pb == "lasso":
         pb = LASSO
-    elif pb.lower() == "logreg":
+    elif pb == "logreg":
         pb = LOGREG
         if set(y) - set([-1.0, 1.0]):
             raise ValueError(
                 "y must contain only -1. or 1 values. Got %s " % (set(y)))
-    elif pb.lower() == "grouplasso":
+    elif pb == "grouplasso":
```

Comment:
same here, it's globally a good idea to do it but it does not belong in this PR. Keep the diff as simple as possible.